utils/lazy-pages-fuzzer/src/generate/mem_accesses.rs

```diff
@@ -0,0 +1,149 @@
+// This file is part of Gear.
+
+// Copyright (C) 2021-2024 Gear Technologies Inc.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+use arbitrary::Unstructured;
+use derive_more::{Display, Error, From};
+use gear_wasm_instrument::parity_wasm::elements::{External, Instruction, Module};
+
+use crate::OS_PAGE_SIZE;
+
+#[derive(Debug, Clone)]
+pub struct InjectMemoryAccessesConfig {
+    pub max_accesses_per_func: usize,
+}
+
+impl Default for InjectMemoryAccessesConfig {
+    fn default() -> Self {
+        InjectMemoryAccessesConfig {
+            max_accesses_per_func: 10,
+        }
+    }
+}
+
+#[derive(Debug, Display, Error, From)]
+pub enum InjectMemoryAccessesError {
+    #[display(fmt = "No memory imports found")]
+    NoMemoryImports,
+    #[display(fmt = "No code section found")]
+    NoCodeSection,
+    #[display(fmt = "")]
+    Arbitrary(arbitrary::Error),
+}
+
+// TODO: different word size accesses
```

Comment:
Please open for the todo a separate issue

Reply:
Done #4042